odoo/addons/base/models/res_partner.py

```diff
@@ -46,6 +46,28 @@ class FormatAddressMixin(models.AbstractModel):
     _name = "format.address.mixin"
     _description = 'Address Format'
 
+    def _extract_fields_from_address(self, address_line):
+        """
+        Extract keys from the address line.
+        For example, if the address line is "zip: %(zip)s, city: %(city)s.",
+        this method will return ['zip', 'city'].
+        """
+        last_starting_index_field = None
+        address_line_length = len(address_line)
+        fields = []
+
+        for i in range(address_line_length):
+            if address_line[i:i + 2] == '%(':
+                last_starting_index_field = i
+                i = i + 1  # Skip the next character as we have already checked it
+            if address_line[i:i + 2] == ')s' and last_starting_index_field is not None:
+                field = address_line[last_starting_index_field + 2: i]
+                if field:
+                    fields.append(field)
+                last_starting_index_field = None
+                i = i + 1  # Skip the next character as we have already checked it
```

Comment:
as far as I can tell, this should do nothing (i will be assigned to the next value in each loop)

same thing for the one above

you need a "while i < len(address_line_length):" for something to change the place in the iteration

Reply:
in that case, i wanted to jump by two characters so thats why there is `i = i + 1` and the for loop will add another one to `i`, if it was while loop it will be in that `if` this way `i = i + 2`